jbpm-flow-builder/src/main/java/org/jbpm/process/builder/ProcessBuildContext.java

```diff
@@ -31,7 +31,7 @@
     private ProcessDescr processDescr;
     private DialectCompiletimeRegistry dialectRegistry;
 
-    public ProcessBuildContext(final KnowledgeBuilderImpl pkgBuilder,
+    public ProcessBuildContext(final DroolsAssemblerContext pkgBuilder,
```

Comment:
From readability perspective this confuses me (the use of "DroolsAssemblerContext" for a "package builder" field inside ProcessBuilderContext...what?

Reply:
Yes I agree. I'll change the variable name. This needs a bit of an explanation, anyway. 

Here in jBPM an internal (`Impl`) of Drools was used. `KnowledgeBuilderImpl` has a *huge* API surface. We'll have to **deprecate** that. The `Assembler` mechanism should help with that, but it's unfeasible at this point to avoid using all those methods; as an interim measure, I have extracted all and only the methods of `KnowledgeBuilderImpl` that are used in `ProcessBuildContext` (and a bunch of other places) to the [`DroolsAssemblerContext`](https://github.com/kiegroup/drools/pull/2181/files#diff-8536ce258aa279001c9e66571938b099) interface. This interface should disappear in the future, in favor of a more general API. The name `Assembler` comes from the fact that we have now moved all of this processing in an `Assembler`. and `Drools`, because it's still relying on Drools internals.

Reply:
> Here in jBPM

heh, just to be clear: that's no blaming, we were reusing some code from Drools that relied *itself* too much on internal impl details...